content/blog/2024/zoomies.md

```diff
@@ -0,0 +1,464 @@
+but to make the interface more legible
+it renders at a `1920 x 1080`
+resolution by default.
+
+That means we have twice the _pixel density_ --
+we can now fit multiple physical pixels
+inside a single resolution 'pixel'!
+That ratio of rendered pixels to physical pixels
+is called the _device pixel ratio_.
+For my screen, at the default settings,
+I have a device pixel ratio
+(what CSS calls a 'resolution')
+of `2x` or `2dppx` (_dots per `px` unit_).
+
+To phrase it differently,
+the entire operating system is
+_zoomed in 200% by default_
+on this monitor.
+
+CSS adds another layer of pixel abstraction.
+While the CSS `px` unit is generally
+equivalent to a single OS-provided 'pixel' by default,
+there are several ways they can diverge --
+including the zoom/scale options
+that we'll discuss below.
+
+But also,
+rather than having fixed
+real-world dimensions,
+the _CSS pixel_ (`1px`) and _CSS inch_ (`1in`)
+have a fixed _relationship to each other_.
+There are always `96` CSS pixels for every CSS inch.
+But depending on the output media (screen vs print),
+their actual sizes might vary:
+
+- On screens,
+  the `px` acts as an '[anchor unit](https://drafts.csswg.org/css-values/#anchor-unit)',
+  and physical units (like `in`, `cm`, `mm`)
+  are determined relative to that.
+  It might not match a real-word inch,
+  but you can count on the relationship: `1in == 96px`.
+- In print,
+  where physical measurements are more reliable,
+  the 'physical' units act as our anchor --
+  and the CSS pixel unit becomes relative instead
+  (`1px == 1/96in`).
+
+On screens,
+the actual size of a CSS inch
+depends on the screen resolution --
+and may not be anywhere close to a physical inch.
+But once you hit 'print',
+the inch becomes reliable,
+and pixels will resize to fit.
+
+No matter what the result,
+that `1:96` inch-to-px relationship
+(determined by
+[the arm-length of Håkon Wium Lie](https://drafts.csswg.org/css-values/#reference-pixel))
+is always the same.
+We only change which unit is anchored to the media,
+and which one is adjusting to maintain the ratio.
+
+Elika Etemad
+(aka Fantasai)
+covered this in her recent talk at CSS Day:
+[_Standardization Stories_](https://www.youtube.com/watch?v=krh_nb9PdVk).
+
+## Why are there two viewports?
+
+You might have heard about
+_the browser viewport_,
+or even used _viewport units_
+(`vw`/`vi`/etc).
+But browsers actually provide
+_two viewports_:
+
+- The **layout viewport**
+  is the box that we put our web pages in.
+  You can think of it like
+  the parent element of the `<html>` tag.
+  It might overflow (and have scrollbars),
+  but it still has a fixed size
+  based on your browser window.
+  When we make that layout viewport larger,
+  we can fit more things on the page
+  without overflowing.
+- The **visual viewport**
+  is the window we look through
+  to see the page.
+  When we make the visual viewport larger,
+  we can see more of the layout at once.
+
+Those might sound the same,
```

Comment:
Would another simpler example be a typical webpage with vertical scrolling? The layout viewport is the entire page, even what is off screen, and the visual viewport is the part you can scroll to.

Or is that not the same thing?

Reply:
Surprisingly, they are not! The layout viewport (like other CSS boxes) can overflow, and get scrollbars. But it doesn't actually grow with the content. (I'll try and clarify that here)